// cloudmersive-convert-api-client/src/model/DocxTableTableFillMultiRequest.js
/**
 * convertapi
 * Convert API lets you effortlessly convert file formats and types.
 *
 * OpenAPI spec version: v1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.5
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/DocxTableSingleTableFill'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./DocxTableSingleTableFill'));
  } else {
    // Browser globals (root is window)
    if (!root.CloudmersiveConvertApiClient) {
      root.CloudmersiveConvertApiClient = {};
    }
    root.CloudmersiveConvertApiClient.DocxTableTableFillMultiRequest = factory(root.CloudmersiveConvertApiClient.ApiClient, root.CloudmersiveConvertApiClient.DocxTableSingleTableFill);
  }
}(this, function(ApiClient, DocxTableSingleTableFill) {
  'use strict';




  /**
   * The DocxTableTableFillMultiRequest model module.
   * @module model/DocxTableTableFillMultiRequest
   * @version 2.7.4
   */

  /**
   * Constructs a new <code>DocxTableTableFillMultiRequest</code>.
   * Request to update data in a table in a Word DOCX Document
   * @alias module:model/DocxTableTableFillMultiRequest
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>DocxTableTableFillMultiRequest</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/DocxTableTableFillMultiRequest} obj Optional instance to populate.
   * @return {module:model/DocxTableTableFillMultiRequest} The populated <code>DocxTableTableFillMultiRequest</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('InputFileUrl')) {
        obj['InputFileUrl'] = ApiClient.convertToType(data['InputFileUrl'], 'String');
      }
      if (data.hasOwnProperty('InputFileData')) {
        obj['InputFileData'] = ApiClient.convertToType(data['InputFileData'], 'Blob');
      }
      if (data.hasOwnProperty('TablesToFill')) {
        obj['TablesToFill'] = ApiClient.convertToType(data['TablesToFill'], [DocxTableSingleTableFill]);
      }
    }
    return obj;
  }

  /**
   * Optional; Input URL of the document; use BeginEditing to create this
   * @member {String} InputFileUrl
   */
  exports.prototype['InputFileUrl'] = undefined;
  /**
   * Optional; Input Word Document file content for the operation
   * @member {Blob} InputFileData
   */
  exports.prototype['InputFileData'] = undefined;
  /**
   * Tables and datasets to fill into the document
   * @member {Array.<module:model/DocxTableSingleTableFill>} TablesToFill
   */
  exports.prototype['TablesToFill'] = undefined;



  return exports;
}));
