// cloudmersive-convert-api-client/src/model/GetDocxGetFormFieldsResponse.js
/**
 * convertapi
 * Convert API lets you effortlessly convert file formats and types.
 *
 * OpenAPI spec version: v1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.5
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/DocxContentControl', 'model/HandlebarFormField'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./DocxContentControl'), require('./HandlebarFormField'));
  } else {
    // Browser globals (root is window)
    if (!root.CloudmersiveConvertApiClient) {
      root.CloudmersiveConvertApiClient = {};
    }
    root.CloudmersiveConvertApiClient.GetDocxGetFormFieldsResponse = factory(root.CloudmersiveConvertApiClient.ApiClient, root.CloudmersiveConvertApiClient.DocxContentControl, root.CloudmersiveConvertApiClient.HandlebarFormField);
  }
}(this, function(ApiClient, DocxContentControl, HandlebarFormField) {
  'use strict';




  /**
   * The GetDocxGetFormFieldsResponse model module.
   * @module model/GetDocxGetFormFieldsResponse
   * @version 2.7.4
   */

  /**
   * Constructs a new <code>GetDocxGetFormFieldsResponse</code>.
   * Result of running a DocxGetFormFields command
   * @alias module:model/GetDocxGetFormFieldsResponse
   * @class
   */
  var exports = function() {
    var _this = this;




  };

  /**
   * Constructs a <code>GetDocxGetFormFieldsResponse</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/GetDocxGetFormFieldsResponse} obj Optional instance to populate.
   * @return {module:model/GetDocxGetFormFieldsResponse} The populated <code>GetDocxGetFormFieldsResponse</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('Successful')) {
        obj['Successful'] = ApiClient.convertToType(data['Successful'], 'Boolean');
      }
      if (data.hasOwnProperty('ContentControls')) {
        obj['ContentControls'] = ApiClient.convertToType(data['ContentControls'], [DocxContentControl]);
      }
      if (data.hasOwnProperty('HandlebarFormFields')) {
        obj['HandlebarFormFields'] = ApiClient.convertToType(data['HandlebarFormFields'], [HandlebarFormField]);
      }
    }
    return obj;
  }

  /**
   * True if successful, false otherwise
   * @member {Boolean} Successful
   */
  exports.prototype['Successful'] = undefined;
  /**
   * Content controls in the DOCX
   * @member {Array.<module:model/DocxContentControl>} ContentControls
   */
  exports.prototype['ContentControls'] = undefined;
  /**
   * Form fields that comply with the Handlebar style, that is they are surrounded by two curly braces on either side such as \"{{FieldName}}\", in the DOCX
   * @member {Array.<module:model/HandlebarFormField>} HandlebarFormFields
   */
  exports.prototype['HandlebarFormFields'] = undefined;



  return exports;
}));
